src/server/worker/master.go

```diff
@@ -473,6 +506,14 @@ func (a *APIServer) waitJob(pachClient *client.APIClient, jobInfo *pps.JobInfo,
 					if err := a.jobs.ReadWrite(stm).Get(jobInfo.Job.ID, jobPtr); err != nil {
 						return err
 					}
+					if jobInfo.EnableStats {
+						if _, err = pachClient.PfsAPIClient.FinishCommit(ctx, &pfs.FinishCommitRequest{
+							Commit: jobInfo.StatsCommit,
+							Empty:  true,
+						}); err != nil {
+							logger.Logf("error from FinishCommit for stats while timing out job: %+v", err)
```

Comment:
This code path is primarily for stop job not timeouts, but timeouts as well as failed inputs will take this path. Also, the stats commit can get finished before here, so I think this should be filtering on finished commit errors.

Reply:
I'll change the error message and filter out commit already finished errors.